src/software/new_geom/BUILD

```diff
@@ -39,6 +43,28 @@ cc_test(
     ],
 )
 
+cc_test(
+    name = "closest_point_test",
+    srcs = [
+        "util/closest_point_test.cpp",
+    ],
+    deps = [
+        ":new_geom",
```

Comment:
Do we need all of `new_geom` to test this? We should make sure libraries are as small as possible so it's clear what's being tested. If we don't need all of `new_geom` here we should make a `closest_point` library and just test that (and have `new_geom` depend on it).

Reply:
Same goes for `distance_test`, I'm really trying to avoid having a massive monolithic geom library like we currently do.

Reply:
Added `closest_point` and `distance` as separate libraries, and `new_geom`, `closest_point_test`, and `distance_test` depend on them.